src/town_cmd.cpp

```diff
@@ -516,6 +516,37 @@ static void MakeTownHouseBigger(TileIndex tile)
 	if (flags & BUILDING_HAS_4_TILES) MakeSingleHouseBigger(TILE_ADDXY(tile, 1, 1));
 }
 
+/**
+ * Generate cargo for a house, scaled by the current economy scale.
+ * @param t The current town.
+ * @param ct Type of cargo to generate, usually CT_PASSENGERS or CT_MAIL.
+ * @param amount The number of cargo units.
+ * @param stations Available stations for this house.
+ * @param affected_by_recession Is this cargo halved during recessions?
+ */
+static void TownGenerateCargo(Town *t, CargoID ct, uint amount, StationFinder &stations, bool affected_by_recession)
+{
+	uint cargo_amount = amount;
```

Comment:
Why this assignment? You can just name the function parameter `cargo_amount`, or use `amount` through the function.

Reply:
I thought it was forbidden/frowned upon for a function to modify the arguments passed to it. Did I imagine this, or is it language-specific, perhaps?

Reply:
amount is a value type not a reference type here, so there is nothing wrong with modifying it in the function body.
There is no effect on the caller scope.